packages/block-editor/src/utils/show-block-removal-warning.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Modal,
+	Button,
+	__experimentalHStack as HStack,
+} from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+
+export function showBlockRemovalWarning( blockName ) {
```

Comment:
Hmm, I'm wondering about the mixed type return value. The function name, at least in my mind, implies a bool, but I see `blockName` is handy to have to display the removal prompt content.

Could we rather pass down a list of IDs and return those that required removal. That means we could memoize in one spot too. E.g.,

`getBlockNamesWithRemovalWarnings( clientIds ) // => [ 'core/query' ]`

```js
/**
 * 
 * @param {Array} clientIds 
 * @returns {Array}
 */
const getBlockNamesWithRemovalWarnings = ( clientIds ) =>
    clientIds
        .map( getBlockName )
        .filter( ( blockName ) => {
        return blockName === 'core/query' || blockName === 'core/post-content';
        } );
```

Would that work?